File: src/types/sensor.ts
// Matches the backend SensorResponse schema
export interface SensorReading {
  id: number;
  temperature: number;
  humidity: number;
  soil_moisture: number;
  nitrogen: number;
  phosphorus: number;
  potassium: number;
  ph: number;
  ec: number | null;
  created_at: string;
}

export interface IrrigationStatus {
  irrigation_needed: boolean;
  water_amount_liters: number;
  message: string;
}

export interface IrrigationHistoryItem extends IrrigationStatus {
  id: number;
  created_at: string;
}

File: backend/app/models/sensor.py
from sqlalchemy import Column
from sqlalchemy import Integer
from sqlalchemy import Float
from sqlalchemy import DateTime

from datetime import datetime

from app.database.database import Base


class SensorReading(Base):
    __tablename__ = "sensor_readings"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    temperature = Column(Float)
    humidity = Column(Float)

    soil_moisture = Column(Float)

    ph = Column(Float)

    nitrogen = Column(Float)
    phosphorus = Column(Float)
    potassium = Column(Float)

    created_at = Column(
        DateTime,
        default=datetime.utcnow
    )